common/components/GasSlider/components/SimpleGas.tsx

```diff
@@ -3,22 +3,41 @@ import Slider from 'rc-slider';
 import translate from 'translations';
 import { gasPriceDefaults } from 'config/data';
 import FeeSummary from './FeeSummary';
+import { Spinner } from 'components/ui';
+import { CSSTransition } from 'react-transition-group';
+import { State as NetworkState } from 'reducers/transaction/network';
 import './SimpleGas.scss';
 
 interface Props {
   gasPrice: string;
+  gasEstimationStatus: NetworkState['gasEstimationStatus'];
   changeGasPrice(gwei: string): void;
 }
 
 export default class SimpleGas extends React.Component<Props> {
   public render() {
-    const { gasPrice } = this.props;
-
+    const { gasPrice, gasEstimationStatus } = this.props;
+    const estimatingGas = gasEstimationStatus === 'PENDING' ? true : false;
+    const estimatingGasTimeout = gasEstimationStatus === 'TIMEOUT' ? true : false;
     return (
       <div className="SimpleGas row form-group">
-        <div className="col-md-12">
+        <div className="col-md-12 SimpleGas-title">
           <label className="SimpleGas-label">{translate('Transaction Fee')}</label>
+          <div className="SimpleGas-flex-spacer" />
+          <CSSTransition in={estimatingGas} timeout={300} classNames="fade">
+            <div className={`SimpleGas-estimating small ${estimatingGas ? 'active' : ''}`}>
+              Setting gas limit
+              <Spinner />
+            </div>
+          </CSSTransition>
         </div>
+        {estimatingGasTimeout && (
+          <div className="col-md-12 prompt-toggle-gas-limit">
+            <p className="small">
+              Couldn't set gas limit, try setting manually in advanced settings
```

Comment:
I'd recommend switching nodes, or telling them to wait before asking a user to set the gas limit manually. Maybe 1% of users will know what to do in this situation. But for 99% of cases, we should probably just be able to set it to 21000 right?

Reply:
Token transfers are a common usecase, and 21000 would not suffice there.

Reply:
Right, I meant the behavior could be `if (!data) return 21000`, and only bug the user about estimate failures if they have data (from token or otherwise.)